x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -567,7 +567,7 @@ public Security(Settings settings) {
         this.settings = settings;
         // TODO this is wrong, we should only use the environment that is provided to createComponents
         this.enabled = XPackSettings.SECURITY_ENABLED.get(settings);
-        this.systemIndices = new SecuritySystemIndices();
+        this.systemIndices = new SecuritySystemIndices(settings);
```

Comment:
Besides the fact that some `Settings` passed around are not complete, the caller of `SecuritySystemIndices#getSystemIndexDescriptor` (which returns system index descriptors) also takes a `Settings` parameter.
It's sort of a mess.

> Also, this is a minor concern since it is not actually an issue we have tons of other usages. Feel free to completely ignore this comment and we can address when fixing the TODO :)

I'll go with that if you don't mind.